src/book/02-system/01-foundations/01-overview.mdx

```diff
@@ -22,3 +22,42 @@ After motions is hardware IO. The hardware will output the commands and input th
+| `sensortest`         | Real robot sensors and NUsight                                                                        | Used to view the robot's position and orientiation in NUsight, often to test odometry. Can also be used to check for broken servos without the robot moving, because servo errors and warnings will be printed to the terminal.                                         |
+| `visualmesh_fake`    | Fake camera and vision modules                                                                        | Used to test the vision system, such as detection of balls and goals, in a simulated environment.                                                                                                                                                                       |
+| `visualmesh`         | Real robot sensors and camera with vision modules                                                     | Used to test the vision system on a real robot.                                                                                                                                                                                                                         |
+| `webots_keyboard`    | `keyboardwalk` but with Webots instead of real robot sensors                                          | Used for easily testing motions in Webots.                                                                                                                                                                                                                              |
+| `webots`             | Webots connection module                                                                              | Used for testing the connection between the robot code and Webots.                                                                                                                                                                                                      |
+| `webotsrobocup`      | Identical to `robocup` but with webots instead of real robot sensors                                  | Used to play RoboCup games in the Webots simulator.                                                                                                                                                                                                                     |
+
+In most cases, `platform::darwin::HardwareIO` and `platform::Webots` can be used interchangably depending on if you are running on the real robot or the simulated robot in Webots. In some cases `platform::darwin::HardwareSimulator` can also be used, but this simulated robot is very basic and does not consider gravity - this may be useful if you want to see a motion run on the robot without the robot falling down.
```

Comment:
Maybe mention that `HardwarIO` will crash if not run on real robot, whilst `Webots` and `HardwareSimulator` will be mostly fine.

Reply:
You would also need to mention that `Webots` (and `Gazebo`) will only work if the appropriate simulator and controller/plugin are loaded correctly as well. `HardwareSimulator` should, in general, always run

Reply:
I put more explanation of the platforms but tbh this should be on a different page, like a platform page.

Reply:
I'm thinking leave it here for now and then move it over when a platform page is created? I'll add it to my list ✍️